pkg/model/dev_test.go

```diff
@@ -1574,3 +1574,157 @@ func TestBuildInfo_GetDockerfilePath(t *testing.T) {
 		})
 	}
 }
+
+func TestExpandBuildArgs(t *testing.T) {
+	t.Setenv("KEY", "VALUE")
+	tests := []struct {
+		name               string
+		buildInfo          *BuildInfo
+		previousImageBuilt map[string]string
+		expected           *BuildInfo
+	}{
+		{
+			name:               "no build args",
+			buildInfo:          &BuildInfo{},
+			previousImageBuilt: map[string]string{},
+			expected:           &BuildInfo{},
+		},
+		{
+			name: "only buildInfo without expanding",
+			buildInfo: &BuildInfo{
+				Args: BuildArgs{
+					{
+						Name:  "KEY",
+						Value: "VALUE",
+					},
+				},
+			},
+			previousImageBuilt: map[string]string{},
+			expected: &BuildInfo{
+				Args: BuildArgs{
+					{
+						Name:  "KEY",
+						Value: "VALUE",
+					},
+				},
+			},
+		},
+		{
+			name: "only buildInfo expanding",
+			buildInfo: &BuildInfo{
+				Args: BuildArgs{
+					{
+						Name:  "KEY",
+						Value: "$KEY",
+					},
+				},
+			},
+			previousImageBuilt: map[string]string{},
+			expected: &BuildInfo{
+				Args: BuildArgs{
+					{
+						Name:  "KEY",
+						Value: "VALUE",
+					},
+				},
+			},
+		},
+		{
+			name:      "only previousImageBuilt",
+			buildInfo: &BuildInfo{},
+			previousImageBuilt: map[string]string{
+				"KEY": "VALUE",
+			},
+			expected: &BuildInfo{
+				Args: BuildArgs{
+					{
+						Name:  "KEY",
+						Value: "VALUE",
+					},
+				},
+			},
+		},
+		{
+			name: "buildInfo args and previousImageBuilt without expanding",
+			buildInfo: &BuildInfo{
+				Args: BuildArgs{
+					{
+						Name:  "KEY",
+						Value: "VALUE",
+					},
+				},
+			},
+			previousImageBuilt: map[string]string{
+				"KEY2": "VALUE",
```

Comment:
```suggestion
				"KEY2": "VALUE2",
```

Nit, but it can helps to avoid confusion to understand if the `VALUE` can come for `KEY` or `KEY2`. Apply to other cases where the 2 keys are used